service/history/task/cross_cluster_task.go

```diff
@@ -208,47 +265,396 @@ func newCrossClusterTaskBase(
+
+	t.scope.IncCounter(metrics.TaskFailuresPerDomain)
+	if _, ok := err.(*persistence.CurrentWorkflowConditionFailedError); ok {
+		t.logger.Error("More than 2 workflow are running.", tag.Error(err), tag.LifeCycleProcessingFailed)
+		return nil
+	}
+	t.logger.Error("Fail to process task", tag.Error(err), tag.LifeCycleProcessingFailed)
+	return err
 }
 
 func (t *crossClusterSourceTask) RetryErr(
 	err error,
 ) bool {
-	panic("Not implement")
+	return err != errWorkflowBusy && err != ErrTaskPendingActive && !common.IsContextTimeoutError(err)
 }
 
 func (t *crossClusterSourceTask) IsReadyForPoll() bool {
 	t.Lock()
 	defer t.Unlock()
 
 	return t.state == ctask.TaskStatePending &&
-		(t.processingState == processingStateInitialized || t.processingState == processingStateResponseRecorded)
+		(t.processingState == processingStateInitialized ||
+			t.processingState == processingStateResponseRecorded)
+}
+
+// GetCrossClusterRequest returns a CrossClusterTaskRequest and error if there's any
+// If the returned error is not nil:
+// - there might be error while loading the request, we can retry the function call
+// - task may be invalidated, in which case caller should submit the task for processing
+//   so that a new task can be created.
+// If both returned error and request are nil:
+// - there's nothing need to be done for the task, task already acked and is not available
+//   for polling again
+// If the returned request is not nil
+// - the request can be returned to the target cluster
+func (t *crossClusterSourceTask) GetCrossClusterRequest() (request *types.CrossClusterTaskRequest, retError error) {
+	t.Lock()
+	defer func() {
+		if retError == nil && request == nil {
+			t.state = ctask.TaskStateAcked
+		}
+		t.Unlock()
+	}()
+
+	if !t.isValidLocked() {
+		return nil, errors.New("task invalidated")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), taskDefaultTimeout)
+	defer cancel()
+
+	taskInfo := t.GetInfo().(*persistence.CrossClusterTaskInfo)
+	_, mutableState, release, err := loadWorkflowForCrossClusterTask(ctx, t.executionCache, taskInfo, t.shard.GetMetricsClient(), t.logger)
+	if err != nil || mutableState == nil {
+		return nil, err
+	}
+	defer func() { release(retError) }()
+
+	request = &types.CrossClusterTaskRequest{
+		TaskInfo: &types.CrossClusterTaskInfo{
+			DomainID:            t.GetDomainID(),
+			WorkflowID:          t.GetWorkflowID(),
+			RunID:               t.GetRunID(),
+			TaskID:              t.GetTaskID(),
+			VisibilityTimestamp: common.Int64Ptr(t.GetVisibilityTimestamp().UnixNano()),
+		},
+	}
+
+	switch t.GetTaskType() {
+	case persistence.CrossClusterTaskTypeStartChildExecution:
+		attributes, taskState, err := t.getRequestForStartChildExecution(ctx, taskInfo, mutableState)
+		if err != nil || attributes == nil {
+			return nil, err
+		}
+		request.TaskInfo.TaskType = types.CrossClusterTaskTypeStartChildExecution.Ptr()
+		request.TaskInfo.TaskState = int16(taskState)
```

Comment:
nit: you can consider moving the common assignments like this to the end.

Reply:
sure. updated.